code/_onclick/adjacent.dm

```diff
@@ -72,6 +72,9 @@
 		return FALSE
 	if(loc.Adjacent(neighbor,target = neighbor, mover = src))
 		return TRUE
+	for(var/turf/T in locs) //this is to handle multi tile objects

```

Comment:
Does the likely expensive check twice for negative results for every single tile objects now.
Does not handle  checks against multi-tile objects only from.
Does not change 9999 uses of get turf as the adjacency check in the current code.
There's more issues with this but these are the major ones.